.github/workflows/ruby.yml

```diff
@@ -0,0 +1,66 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+
+    runs-on: ['ubuntu-latest', 'windows-latest']
```

Comment:
Runs on doesn't accept a matrix here.  This needs an `os` map and then reference with `${{ matrix.os }}`  e.g. https://github.com/glennsarti/puppetfile-resolver/blob/master/.github/workflows/ci.yml#L19-L29